src/nu/validator/htmlparser/io/Encoding.java

```diff
@@ -106,11 +329,12 @@
             Charset cs = entry.getValue();
             String name = toNameKey(cs.name());
             String canonName = toAsciiLowerCase(cs.name());
-            if (!isBanned(name)) {
+            if (!isBanned(stripDashAndUnderscore(name))) {
```

Comment:
> It's a bit unclear to me where this patch is presently going in terms of compliance if `isBanned` stays.

I guess I see this as another case where (with this patch applied) the implementation is functionally equivalent to the spec prose. But I agree it would be better to make it as explicitly more aligned with the spec prose as possible — for example, by rewriting the code to have an allow list rather than a blocklist.

So I can make some time to work on the code in this branch do make it like that.